Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -330,4 +330,163 @@ public TablesResource.InsertRequest makeInsertReq()
+    ///   if it is not. All data in "my_table" will be deleted if the user accepts.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Remove-BqTable -DatasetId "my_dataset" -Table "my_table" -Force</code>
+    ///   <para>This will remove "my_table" and all of its data.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "BqTable", SupportsShouldProcess = true)]
+    
+    public class RemoveGcbqTable : BqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByValue = "ByValue";
+            public const string ByDatasetObject = "ByDatasetObject";
+            public const string ByObject = "ByObject";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for tables in. If not set via PowerShell parameter processing, it will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValue)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        override public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the table that you want to get a descriptor object for.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByValue)]
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        public string Table { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The ID of the dataset to search. Can be a string, a Dataset, a DatasetReference, or a DatasetsData object.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ByValue)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to search. This field takes Dataset or DatasetRefrence objects.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        public DatasetReference Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be inserted. Also takes TableReferences.
```

Comment:
Removed, not inserted. Mention Table object in the comments, not TableReference. TableReference is the actual type of the parameter, and is visible to the user. Table is not visible so it needs the comment.